### PR TITLE
Ref issue 158 add more unit tests and comments for Collective class.

### PR DESCRIPTION
**Pull request by tai271828**
This commit add four more unit tests related to test Collective class, including:
- test_partition_edgecut
- test_partition_part
- test_splitted_ncell_by_clnds
- test_splitted_ncell_by_clfcs

The attributes they want to test are used by both of the legacy and the new "look-up Table" method for splitting blocks in parallel mode, so these unit tests not only make the Table method more robust, but also benefit the development of Ref issue #158. When porting the legacy method to the look-up Table method, these test cases could be always useful.

Thus, this commit is a part of the issue #158.

Besides, more comments for the other unit tests related to the Collective class were also added to the better maintenance.

### CODE REVIEW

**Comment by tai271828** [comment]
CI failed when running this environment

```
    - os: osx
      osx_image: xcode7.3
      language: cpp
      compiler: clang
      env: TRAVIS_PYTHON_VERSION="3.5"
```

and the error messages are:

```
======================================================================
FAIL: To ensure the partitioner returns expected values.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/solvcon/solvcon/solvcon/tests/test_domain.py", line 61, in test_partition_edgecut
    self.assertEqual(self.dom.edgecut, 16)
AssertionError: 17 != 16
======================================================================
FAIL: To ensure the partitioner returns expected values.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/solvcon/solvcon/solvcon/tests/test_domain.py", line 83, in test_partition_part
    self.assertTrue(np.array_equal(self.dom.part, answer_part))
AssertionError: False is not true

```

QUESTION:
Should different compilers result in different partition results in this case?

**Comment by yungyuc** [comment]
My guess is SCOTCH (not METIS) behaves differently on linux and osx.

I think we need a much smaller and predictable fixture.  Figuring out why get_sample_neu() is splitted differently on linux and osx would be too much work.

File: solvcon/tests/test_domain.py [hunk 1]
```diff
@@ -43,10 +45,88 @@ def test_group(self):
...
+    def test_partition_edgecut(self):
+        """
+        To ensure the partitioner returns expected values.
+
+        SOLVCON uses METIS to be the partitioner,
+        and this could be replaced or change anytime.
+        Please refer to mesh.pyx:Mesh:partition for more details.
+
+        This test assumes two presumption:
+            1. partitioner is METIS.
```
Comment (yungyuc) [inline solvcon/tests/test_domain.py:57]: We use SCOTCH actually.  SCOTCH has a METIS compatibility layer.

**Comment by yungyuc** [comment]
Feel free to reopen the PR if you want to update it.

**Pull request closed without merging**